hoot-core/src/main/cpp/hoot/core/ops/CalculateStatsOp.cpp

```diff
@@ -552,7 +549,7 @@ void CalculateStatsOp::_generateFeatureStats(boost::shared_ptr<const OsmMap>& ma
   double totalFeatures = 0.0;
   totalFeatures =
     _applyVisitor(
-      map, FilteredVisitor(e_criterion->clone(), _getElementVisitorForFeatureType(featureType)));
+      map, FilteredVisitor(*(e_criterion->clone()), *(_getElementVisitorForFeatureType(featureType))));
```

Comment:
I think that these additions shouldn't be necessary.  Filtered visitor has a constructor like so:
```
FilteredVisitor(ElementCriterionPtr criterion, ConstElementVisitorPtr visitor)
```
And `ElementCriterion::clone` returns an `ElementCriterionPtr` while `_getElementVisitorForFeatureType` returns a `ConstElementVisitorPtr`.  So the code might look like this:
```
      map, FilteredVisitor(criterion->clone(), _getElementVisitorForFeatureType(featureType)));
```
Also for the sake of cleanliness, `e_criterion` is actually no longer needed because `criterion` is actually of the same type.

Reply:
Nice catch. I made this one change before I cut to the chase & added
```
FilteredVisitor(ElementCriterionPtr criterion, ConstElementVisitorPtr visitor)
```

Reply:
I'll push an update in a few minutes.